### PR TITLE
[consensus] switch from concurrent event processor to sequential

### PR DESCRIPTION
**Pull request by zekun000**
## Motivation

Continued from https://github.com/libra/libra/pull/579, this PR we switch from multiple concurrent tasks to one single task polling new events.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Y

## Test Plan

Existing tests.

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/libra/website, and link to your PR here.)

### CODE REVIEW

File: consensus/src/chained_bft/event_processor.rs [hunk 1]
```diff
@@ -677,75 +508,257 @@ impl<T: Payload> EventProcessor<T> {
...
-    /// 3. Prune the tree.
-    async fn process_commit(
-        &self,
-        committed_block: Arc<Block<T>>,
-        finality_proof: LedgerInfoWithSignatures<Ed25519Signature>,
-    ) {
-        // First make sure that this commit is new.
-        if committed_block.round() <= self.block_store.root().round() {
+impl<T: Payload> EventProcessor<T> {
+    pub fn new(
```
Comment (dmitri-perelman) [inline consensus/src/chained_bft/event_processor.rs]: Why did you move some of the functions to a separate `impl` block? Not that I have something against it, just trying to understand the changes (and it makes the review process more complicated because the UI shows it as new code so it's hard to see the real differences).

Comment (zekun000) [reply consensus/src/chained_bft/event_processor.rs]: I revert this change to make this PR cleaner, I'll create a separate PR for re-organizing.

**Comment by andll** [comment]
Since we now have single task that process all messages, we could have `enum EventProcessorMessage` and single receiver for this message type, instead of `select`.
This would simplify adding new message types (no need to introduce new channel, just add one more element to enum)

**Comment by huitseeker** [comment]
yay! :tada:
